ivy/container/gradients.py

```diff
@@ -1,9 +1,320 @@
+from typing import Optional, Union, List, Dict
+
 # local
+import ivy
 from ivy.container.base import ContainerBase
 
-# ToDo: implement all methods here as public instance methods
-
 
 # noinspection PyMissingConstructor
 class ContainerWithGradients(ContainerBase):
-    pass
+    @staticmethod
+    def static_optimizer_update(
+        ws,
+        effective_grads,
+        lr,
+        inplace=None,
+        stop_gradients=True,
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+    ) -> ivy.Container:
+
+        return ContainerBase.multi_map_in_static_method(
+            "optimizer_update",
+            ws,
+            effective_grads,
+            lr,
+            inplace=inplace,
+            stop_gradients=stop_gradients,
+            key_chains=key_chains,
+            to_apply=to_apply,
+            prune_unapplied=prune_unapplied,
+            map_sequences=map_sequences,
+        )
+
+    def optimizer_update(
+        self: ivy.Container,
+        ws,
```

Comment:
`ws` should be removed here, `self` is replacing `ws`, likewise for the other instance methods. These should actually be causing the tests to fail, but maybe they are not yet implemented. Could you please verify that the container instance methods are all working correctly before requesting the next review? :)

Reply:
totally missed this, yes will do. Thanks.